Source/SwiftLintFramework/Models/Example.swift

```diff
@@ -0,0 +1,71 @@
+/// Captures code and context information for an example of a triggering or
+/// non-triggering style
+public struct Example {
+    public var code: String
+
+    // file and line are optional because we need to conform to
+    // Codable, and StaticString isn't Codable, so we just ignore
```

Comment:
How do we use `Example`'s `Codable` conformance?

Reply:
It looked like `Codable` conformance of rules was strictly for reading and writing caches on disk, and that didn’t seem applicable when testing. Unfortunately, the file has to be stored as a `StaticString` in order to pass to XCTest stuff, which limits our options unless you want to entertain an investigation into unsafe bit casting.

Reply:
Hmmm, I don’t think we should be encoding the examples for caching purposes. I don’t see why it’s being done that way now.

Reply:
The cascade goes Example → RuleDescription → StyleViolation → FileCacheEntry → FileCache and then all hell breaks loose 😓

Reply:
…but maybe it would be possible to store, not the full rule, but just the identifier, in the cache?

Reply:
☝🏻ended up doing that